Tests/GitHubRepositories.tests.ps1

```diff
@@ -290,6 +290,72 @@ try
         }
     }
 
+    Describe 'GitHubRepositories\New-GitHubRepositoryFromTemplate' {
+        BeforeAll -ScriptBlock {
+            $templateRepoName = ([Guid]::NewGuid().Guid)
+            $testGitIgnoreTemplate=(Get-GitHubGitIgnore)[0]
+            $testLicenseTemplate=(Get-GitHubLicense)[0].key
+
+            $newGitHubRepositoryParms = @{
+                RepositoryName = $templateRepoName
+                Description = $defaultRepoDesc
+                GitIgnoreTemplate = $testGitIgnoreTemplate
+                LicenseTemplate = $testLicenseTemplate
+                IsTemplate = $true
+            }
+            $templateRepo = New-GitHubRepository @newGitHubRepositoryParms
+        }
+
+        Context -Name 'When creating a public repository from a template' -Fixture {
+            BeforeAll -ScriptBlock {
+                $repoName = ([Guid]::NewGuid().Guid)
+                $newRepoDesc = 'New Repo Description'
+                $newGitHubRepositoryFromTemplateParms = @{
+                    RepositoryName = $repoName
+                    OwnerName = $script:ownerName
+                    TemplateOwnerName = $templateRepo.owner.login
+                    TemplateRepositoryName = $templateRepoName
+                    Description = $newRepoDesc
+                }
+                $repo = New-GitHubRepositoryFromTemplate @newGitHubRepositoryFromTemplateParms
+            }
+
+            It 'Should return an object of the correct type' {
+                $repo | Should -BeOfType PSCustomObject
```

Comment:
Clearly before the pipeline work, but you'll want to also check `$repo.PSObject.TypeNames[0] | Should -Be 'GitHub.Repository` as well.